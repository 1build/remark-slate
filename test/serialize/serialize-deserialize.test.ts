import unified from 'unified';
import parse from 'remark-parse';
import plugin, { serialize } from '../../src'

const mdown = `a
<br>

[https://a.com](https://a.com)
<br>

e
`

test('Deserialize', done => {
  unified()
    .use(parse)
    .use(plugin)
    .process(mdown, (_, file) => {
      //@ts-ignore
      const res = file.result
      expect(res).toMatchSnapshot()
      //@ts-ignore
      expect(res.map((v) => serialize(v)).join('')).toEqual(mdown)
      done()
    });
})

