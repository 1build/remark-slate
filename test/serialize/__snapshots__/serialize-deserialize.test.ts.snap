// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deserialize 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "text": "a
",
      },
      Object {
        "children": Array [
          Object {
            "text": "",
          },
        ],
        "type": "paragraph",
      },
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [
          Object {
            "text": "https://a.com",
          },
        ],
        "link": "https://a.com",
        "type": "link",
      },
      Object {
        "text": "
",
      },
      Object {
        "children": Array [
          Object {
            "text": "",
          },
        ],
        "type": "paragraph",
      },
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "text": "e",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`Deserialize 2`] = `
"a
<br>

[https://a.com](https://a.com)

e
"
`;
